File: KrishiVoice/fix_useAssistant.js
const fs = require('fs');
const file = '/Users/aman/Desktop/KrishiVoice/KrishiVoice/src/hooks/useAssistant.js';
const lines = fs.readFileSync(file, 'utf8').split('\n');

const startIdx = lines.findIndex((l, i) => l.trim() === 'export { CROP_DATABASE }' && lines[i+1] && lines[i+1].includes('Rice: {'));

if (startIdx !== -1) {
  let endIdx = -1;
  for (let i = startIdx + 1; i < lines.length; i++) {
    if (lines[i].trim() === 'export { CROP_DATABASE };' || 
        (lines[i].trim() === '};' && lines[i+1] && lines[i+1].trim() === '' && lines[i+2] && lines[i+2].includes('const CROP_INTENTS'))) {
      endIdx = i;
      break;
    }
  }
  
  if (endIdx !== -1) {
    const fixedLines = [
      ...lines.slice(0, startIdx),
      'export { CROP_DATABASE };',
      ...lines.slice(endIdx + 1)
    ];
    fs.writeFileSync(file, fixedLines.join('\n'));
    console.log('Fixed useAssistant.js successfully!');
  } else {
    console.log('Could not find end index.');
  }
} else {
  console.log('Could not find start index.');
}

File: KrishiVoice/fix_useAssistant.py
import sys

file_path = '/Users/aman/Desktop/KrishiVoice/KrishiVoice/src/hooks/useAssistant.js'

with open(file_path, 'r') as f:
    lines = f.readlines()

start_idx = -1
for i, line in enumerate(lines):
    if line.strip() == 'export { CROP_DATABASE }' and i + 1 < len(lines) and 'Rice: {' in lines[i+1]:
        start_idx = i
        break

if start_idx != -1:
    end_idx = -1
    for i in range(start_idx + 1, len(lines)):
        if lines[i].strip() == 'export { CROP_DATABASE };' or (lines[i].strip() == '};' and i + 2 < len(lines) and 'const CROP_INTENTS' in lines[i+2]):
            end_idx = i
            break
    
    if end_idx != -1:
        new_lines = lines[:start_idx] + ['export { CROP_DATABASE };\n'] + lines[end_idx+1:]
        with open(file_path, 'w') as f:
            f.writelines(new_lines)
        print("Fixed useAssistant.js successfully!")
    else:
        print("Could not find end index.")
else:
    print("Could not find start index.")
